tests/test_connections/test_ledger/test_ledger.py

```diff
@@ -96,3 +116,112 @@ async def test_not_hanging(self) -> None:
 
         # the `receive_task` should be done, and not await for the `blocking_task`
         assert receive_task.done(), "Receive task is blocked by blocking task!"
+
+
+class TestLedgerConnectionWithMultiplexer(BaseSkillTestCase):
+    """Test `LedgerConnection` class, using the multiplexer, via the `BaseSkillTestCase`."""
+
+    path_to_skill = Path(ROOT_DIR, "packages", "valory", "skills", "abstract_abci")
+    abci_handler: ABCIHandler
+    logger: logging.Logger
+    abci_dialogues: AbciDialogues
+    running_loop: asyncio.AbstractEventLoop
+    thread_loop: Thread
+    _multiplexer: Multiplexer
+    ledger_connection: LedgerConnection
+    make_ledger_connection_callable: Callable = make_ledger_api_connection
+
+    @classmethod
+    def setup(cls, **kwargs: Any) -> None:
+        """Setup the test class."""
+        super().setup()
+        cls.abci_handler = cast(ABCIHandler, cls._skill.skill_context.handlers.abci)
+        cls.logger = cls._skill.skill_context.logger
+
+        cls.abci_dialogues = cast(
+            AbciDialogues, cls._skill.skill_context.abci_dialogues
+        )
+
+        # set up a multiplexer with the required ledger connection
+        cls.running_loop = asyncio.new_event_loop()
+        cls.thread_loop = Thread(target=cls.running_loop.run_forever)
+        cls.thread_loop.start()
+        cls._multiplexer = Multiplexer(
+            [cls.make_ledger_connection_callable()], loop=cls.running_loop
+        )
+        cls._multiplexer.connect()
+        # the ledger connection's connect() is called by the multiplexer
+        # once a connection is ready, `receive()` is called by the multiplexer
+        cls.ledger_connection = cast(LedgerConnection, cls._multiplexer.connections[0])
+
+    @classmethod
+    def teardown(cls) -> None:
+        """Tear down the multiplexer."""
```

Comment:
hm I think also need to call teardown on super, but check

Reply:
`BaseSkillTestCase` does not have a `teardown`.